src/openweathermap.js

```diff
@@ -146,17 +146,19 @@ function populateCurrentUI()
       let sunset = w.getSunsetDate();
       let lastBuild = new Date();
 
-      // Is sunset approaching before the sunrise?
       let ms = lastBuild.getTime();
-      if(sunrise.getTime() - ms > sunset.getTime() - ms)
-      {
+      if (ms < sunrise.getTime()) {
+        this.topBoxSunIcon.set_gicon(this.getGIcon("daytime-sunrise-symbolic"));
+        this.topBoxSunInfo.text = w.displaySunrise(this);
+      }
+      else if (ms > sunrise.getTime() && ms < sunset.getTime()) {
```

Comment:
Kinda mixing bracket styles here, project style seems to be...

```suggestion
      if (ms < sunrise.getTime())
      {
        this.topBoxSunIcon.set_gicon(this.getGIcon("daytime-sunrise-symbolic"));
        this.topBoxSunInfo.text = w.displaySunrise(this);
      }
      else if (ms > sunrise.getTime() && ms < sunset.getTime())
      {
```